src/test/java/com/artipie/rpm/RpmTest.java

```diff
@@ -48,6 +62,64 @@
      */
     private static final String REPOMD = "repodata/repomd.xml";
 
+    /**
+     * Abc test rmp file.
+     */
+    private static final Path ABC =
+        Paths.get("src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm");
+
+    /**
+     * Libdeflt test rmp file.
+     */
+    private static final Path LIBDEFLT =
+        Paths.get("src/test/resources-binary/libdeflt1_0-2020.03.27-25.1.armv7hl.rpm");
+
+    @Test
+    void updatesSameRepoTwiceSuccessfully() throws IOException {
+        final Storage storage = new InMemoryStorage();
+        final Rpm repo =  new Rpm(
+            storage, StandardNamingPolicy.SHA256, Digest.SHA256, true
+        );
+        RpmTest.putFilesInStorage(storage);
+        repo.batchUpdate(Key.ROOT).blockingAwait();
+        repo.batchUpdate(Key.ROOT).blockingAwait();
+        MatcherAssert.assertThat(
+            storage.exists(new Key.From(RpmTest.REPOMD)).join(),
+            new IsEqual<>(true)
+        );
+    }
+
+    @Test
+    void updatesSameRepoSimultaneously() throws Exception {
+        final Storage storage = new InMemoryStorage();
+        final Rpm repo =  new Rpm(
+            storage, StandardNamingPolicy.SHA1, Digest.SHA256, true
+        );
+        RpmTest.putFilesInStorage(storage);
+        final Scalar<Boolean> task = new Unchecked<>(
+            () -> {
+                repo.batchUpdate(Key.ROOT).blockingAwait();
```

Comment:
@olenagerasimova I'd use `CountDownLatch` here to be sure that all threads are waiting before starting update